# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmValidate.cpp
# Commit: 7805ddfe2bfa
# Full Hash: 7805ddfe2bfa83383dc422ed1e8ac70bda627a89
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-19 10:01:16
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Rename TypeCode/ValType/RefType::Any to Extern. r=lth
#   
#   This commit renames the TypeCode/ValType/RefType occurrences of 'any' to 'extern'
#   to match what they actually represent. All public API facing occurrences have
#   already been switched to 'externref' so this should only be aligning the
# ==============================================================================

diff -r 4fa52a0617de -r 7805ddfe2bfa js/src/wasm/WasmValidate.cpp
--- a/js/src/wasm/WasmValidate.cpp	Tue Aug 18 16:58:21 2020 +0000
+++ b/js/src/wasm/WasmValidate.cpp	Tue Aug 18 16:58:32 2020 +0000
@@ -1276,7 +1276,7 @@
         if (!env.gcTypesEnabled()) {
           return iter.unrecognizedOpcode(&op);
         }
-        CHECK(iter.readComparison(RefType::any(), &nothing, &nothing));
+        CHECK(iter.readComparison(RefType::extern_(), &nothing, &nothing));
       }
 #endif
 #ifdef ENABLE_WASM_REFTYPES
@@ -1689,7 +1689,7 @@
             offset = layout.addReference(ReferenceType::TYPE_OBJECT);
             break;
           case RefType::Func:
-          case RefType::Any:
+          case RefType::Extern:
             offset = layout.addReference(ReferenceType::TYPE_WASM_ANYREF);
             break;
         }
@@ -1893,7 +1893,7 @@
   if (elementType == uint8_t(TypeCode::FuncRef)) {
     tableKind = TableKind::FuncRef;
 #ifdef ENABLE_WASM_REFTYPES
-  } else if (elementType == uint8_t(TypeCode::AnyRef)) {
+  } else if (elementType == uint8_t(TypeCode::ExternRef)) {
     if (!refTypesEnabled) {
       return d.fail("expected 'funcref' element type");
     }
@@ -1947,7 +1947,7 @@
     case ValType::Ref:
       switch (type.refTypeKind()) {
         case RefType::Func:
-        case RefType::Any:
+        case RefType::Extern:
           break;
         case RefType::TypeIndex:
 #ifdef WASM_PRIVATE_REFTYPES
@@ -2689,8 +2689,8 @@
             case uint8_t(TypeCode::FuncRef):
               elemType = RefType::func();
               break;
-            case uint8_t(TypeCode::AnyRef):
-              elemType = RefType::any();
+            case uint8_t(TypeCode::ExternRef):
+              elemType = RefType::extern_();
               break;
             default:
               return d.fail(
@@ -2768,7 +2768,7 @@
           return d.fail("failed to read initializer operation");
         }
 
-        RefType initType = RefType::any();
+        RefType initType = RefType::extern_();
         switch (op.b0) {
           case uint16_t(Op::RefFunc):
             initType = RefType::func();